crates/rpc/src/gas_price.rs

```diff
@@ -54,30 +54,39 @@ impl Cached {
                 let tx = Arc::new(tx);
 
                 let inner = self.inner.clone();
-                let eth = self.eth.clone();
+                let gateway = self.gateway.clone();
 
                 g.next = Arc::downgrade(&tx);
 
-                // in general, asking eth_gasPrice seems to be fast enough especially as we already
-                // have a connection open because the EthereumTransport impl is being used for sync
-                // as well.
-                //
-                // it being fast enough, allows us to just coalesce the requests, but also not poll
-                // for fun while no one is using the gas estimation.
+                // Update the gas price from the starknet pending block.
                 tokio::spawn(async move {
-                    let price = match eth.gas_price().await {
-                        Ok(price) => price,
-                        Err(_e) => {
+                    use starknet_gateway_types::reply::MaybePendingBlock;
+                    let gas_price = match gateway
+                        // Don't indefinitely retry as this could block the RPC request.
+                        .block_with_retry(pathfinder_common::BlockId::Pending, Retry::Disabled)
```

Comment:
Why use `block_with_retry` with retry disabled?

Reply:
I suppose I meant `block_with_retry_behaviour(Retry::Disabled)`. However, it could just as well be `block_without_retry` and then it simplifies the public facing API. I'll do that instead.

Reply:
Why not just `block`? :) KISS, YAGNI. To retry or not to retry, ~this is the question~ just let client code decide.

Reply:
Because it 100% must not retry here. And by default our client impl always retries for various reasons like `429` etc.

The rest of the calls by the client should retry. This one must not.

Reply:
Just for the reference, I believe clean design with separation of concerns would look like a simple `Client` and a simple `RetryingClient` wrapper over it (maybe taking specific retry policy per instance or per call). Then the client can decide if retry is needed or not. Not the client being bloated with details if retry should be made.

Similar to IO reader/writer stacks in Java ([source](https://stackoverflow.com/a/35789338)):
```
ByteArrayInputStream bais = new ByteArrayInputStream(responseBytes);
GZIPInputStream gzis = new GZIPInputStream(bais);
InputStreamReader reader = new InputStreamReader(gzis);
BufferedReader in = new BufferedReader(reader);
// ...
```